// [python quickstart](https://grpc.io/docs/quickstart/python.html#run-a-grpc-application)
// python -m grpc_tools.protoc --python_out=. --grpc_python_out=. -I. helloworld.proto

// helloworld.proto
syntax = "proto3";

service dist {
    rpc distSelect(UsrMsg) returns (Select) {}
    rpc distCancel(UsrMsg) returns (Result) {}
    rpc distShow(UsrShow) returns (stream Show) {}
    rpc distHistory(Usr) returns (History) {}
    rpc distComment(QComt) returns (Result) {}
    rpc distReport(UsrReport) returns (Result) {}
}

message QComt {
	string user = 1;
	string commitID = 2;
	string content = 3;
}

message Usr {
    string user = 1;
}

message UsrShow {
    string user = 1;
	string commitID = 2;
	string selected = 3;
}

message UsrMsg {
    string user = 1;
    string commitID = 2;
}

message UsrReport {
	string user = 1;
	string report = 2;
}

message Result {
    int32 result = 1;
}

message Select {
    string commitID = 1;
    string bugzilla = 2;
}

message Show {
    string commitID = 1;
	string detail = 2;
    string bugzilla = 3;
	string comment = 4;
    string user = 5;
	string type = 6;
	string score = 7;
}

message History {
	int32 result = 1;
	string history = 2;
}
